spec/models/move_spec.rb

```diff
@@ -572,4 +572,70 @@
       expect { create(:move, profile: profile) }.to change { profile.reload.updated_at }
     end
   end
+
+  describe '#handle_event_run' do
+    subject(:move) { create(:move) }
+
+    before do
+      allow(Notifier).to receive(:prepare_notifications)
+    end
+
+    context 'when the move has not changed' do
+      it 'returns false' do
+        expect(move.handle_event_run).to eq(false)
+      end
+
+      it 'does not trigger a move notification' do
+        move.handle_event_run
+
+        expect(Notifier).not_to have_received(:prepare_notifications)
+      end
+    end
+
+    context 'when the move status has changed' do
+      before do
+        move.status = 'in_transit'
```

Comment:
what is being tested here? the test looks like the move goes from `in_transit` to `in_transit`?

Reply:
I'm simply testing the plumbing around when the eventable changes it should save the change and hit the has changed branch. I'm not actually interested in the transition itself. The responsibility of this method is to validate when the move has changed (initial state is `requested` (see the [factory](https://github.com/ministryofjustice/hmpps-book-secure-move-api/blob/852c78c2f48f3e7ca4a87834ca3fcee06bd372bc/spec/factories/move.rb#L10)). Would it be

Reply:
For moves, the run handler needs to behave the same as the following https://github.com/ministryofjustice/hmpps-book-secure-move-api/blob/852c78c2f48f3e7ca4a87834ca3fcee06bd372bc/app/services/event_log/move_runner.rb#L46-L51. So we validate that a change leads to a save and a notification when it succeeds/is valid and a different message to the notifier when the status versus other attributes have changed.